examples/city_walking_behaviour/city_walking_behaviour/model.py

```diff
@@ -0,0 +1,729 @@
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_basic_ses4": lambda x: (  # average basic-needs trips for SES=4
+                sum(
+                    agent.basic_needs_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 4
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_basic_ses5": lambda x: (  # average basic-needs trips for SES=5
+                sum(
+                    agent.basic_needs_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 5
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_leisure_ses1": lambda x: (  # average leisure trips for SES=1
+                sum(
+                    agent.leisure_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 1
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_leisure_ses2": lambda x: (  # average leisure trips for SES=2
+                sum(
+                    agent.leisure_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 2
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_leisure_ses3": lambda x: (  # average leisure trips for SES=3
+                sum(
+                    agent.leisure_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 3
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_leisure_ses4": lambda x: (  # average leisure trips for SES=4
+                sum(
+                    agent.leisure_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 4
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+            "avg_leisure_ses5": lambda x: (  # average leisure trips for SES=5
+                sum(
+                    agent.leisure_trips
+                    for agent in x.agents_by_type[Human]
+                    if agent.SES == 5
+                )
+                / len(x.agents_by_type[Human])
+                if x.agents_by_type[Human]
+                else 0
+            ),
+        }
+
+        self.datacollector = DataCollector(model_reporters)
```

Comment:
Maybe we can generate these a bit more elegantly, can you check if something like this works?
```suggestion
        def create_model_reporters():
            """Create a dictionary of model reporters with minimal boilerplate"""
            reporters = {}
            metrics = ['daily_walking_trips', 'work_trips', 'basic_needs_trips', 'leisure_trips']
            
            for metric in metrics:
                for ses in range(1, 6):
                    name = f"avg_{metric.split('_')[0]}_ses{ses}"  # e.g. "avg_walk_ses1"
                    
                    # Create the reporter function
                    reporters[name] = lambda x, m=metric, s=ses: (
                        sum(getattr(agent, m) for agent in x.agents_by_type[Human] if agent.SES == s)
                        / len(x.agents_by_type[Human])
                        if x.agents_by_type[Human]
                        else 0
                    )
            
            return reporters

        # Generate the model_reporters dict
        model_reporters = create_model_reporters()

        self.datacollector = DataCollector(model_reporters)
```